staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -191,40 +229,66 @@ func (r *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	return res, nil
 }
 
-func (a *Authenticator) tokenExpired() bool {
+func (a *Authenticator) credsExpired() bool {
 	if a.exp.IsZero() {
 		return false
 	}
 	return a.now().After(a.exp)
 }
 
 func (a *Authenticator) token() (string, error) {
+	creds, err := a.freshCreds()
+	if err != nil {
+		return "", err
+	}
+	return creds.token, nil
+}
+
+func (a *Authenticator) cert() ([]byte, error) {
```

Comment:
This has the potential to race with key()
```
cert := a.cert()
// key could rotate here
key := a.key()
```

Reply:
Nice catch. Merged GetCert/GetKey and cert/key here.